code/datums/status_effects/debuffs/debuffs.dm

```diff
@@ -690,6 +690,44 @@
 	desc = "You've been zapped with something and your hands can't stop shaking! You can't seem to hold on to anything."
 	icon_state = "convulsing"
 
+/datum/status_effect/repeatedly_electrocute
+	id = "repeated_electrocution"
+	tick_interval = 8 SECONDS
```

Comment:
Actually I thought about this more and I can't help but wonder if this should be a component/element for the object itself?

`/datum/element/periodic_shocks_while_buckled`

(could even be lazy-inited like leaning if there's too many tram rails to do it in init) 

Otherwise we lose context like... surely the tram tracks should *not* shock people if the room is depowered?

Reply:
I don't think the "powered" tram tracks actually care if their area has power or not 🙃 
I can change it so that they do (also for the context of this) though

Reply:
Good news it turns out someone else wrote this 4 years ago